test/configuration_unittest.cc

```diff
@@ -0,0 +1,84 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ protected:
+  void ParseConfiguration(const std::string& configString) {
```

Comment:
The Google style is to use _snake_case_ for variables and parameter names. But you could just name it `input` instead.

Reply:
Changed to `input`.